openedx/core/djangoapps/catalog/utils.py

```diff
@@ -239,6 +242,47 @@ def get_course_runs_for_course(course_uuid):
     else:
         return []
 
+def get_visible_course_runs_for_course(entitlement, course_uuid):
+    """
+    We only want to show courses for a particular entitlement that are:
+
+    1) Currently running or in the future
+    2) A user can enroll in
+    3) A user can upgrade in
+    4) Published
+    """
+    course_runs_for_course = get_course_runs_for_course(str(course_uuid))
+    enrollable_course_runs = []
+
+    # Only show published course runs that can still be enrolled and upgraded
+    now = datetime.datetime.now(UTC)
+    for course_run in course_runs_for_course:
+        # Only courses that have not ended will be displayed
+        run_start = course_run.get('start')
+        run_end = course_run.get('end')
+        is_running = run_start and run_end and dateutil.parser.parse(run_end) > now
+
+        # Only courses that can currently be enrolled in will be displayed
+        enrollment_start = course_run.get('enrollment_start')
+        enrollment_end = course_run.get('enrollment_end')
+        can_enroll = (not enrollment_start or dateutil.parser.parse(enrollment_start) < now
```

Comment:
I think you are missing some parentheses here 
`((not enrollment_start or (dateutil.parser.parse(enrollment_start) < now))
and (not enrollment_end or (dateutil.parser.parse(enrollment_end) > now)))`